andela/activo-api
# #159192785 Create centers

#### What does this PR do?
>- Creates endpoint to create centers
#### Description of Task to be completed?
>- **GIVEN** I have a valid authentication token
>- **WHEN** I `POST` to the `/centers` endpoint
>- **THEN** I will receive a confirmation of the successful creation of a center
#### How should this be manually tested?
>- Run `git fetch origin ft-create-centers-159192785:ft-create-centers-159192785`, then run `git checkout ft-create-centers-159192785`
>- Upgrade your database: `flask db upgrade`
>- Run `flask run`
>- Send a POST request to the endpoint: `localhost:5000/api/v1/centers` with the following payload:
```
{
    "name": "Nairobi",
    "image": {
        "public_id": "cr4mxeqx5zb8rlakpfkg",
        "version": 1372275963,
        "signature": "63bfbca643baa9c86b7d2921d776628ac83a1b6e",
        "width": 864,
        "height": 576,
        "format": "jpg",
        "resource_type": "image",
        "created_at": "2017-06-26T19:46:03Z",
        "bytes": 120253,
        "type": "upload",
        "url": "https://res.cloudinary.com/demo/image/upload/v1372275963/cr4mxeqx5zb8rlakpfkg.jpg",
        "secure_url": "https://res.cloudinary.com/demo/image/upload/v1372275963/cr4mxeqx5zb8rlakpfkg.jpg"
    }
}
```
>- You should receive a success response with the following payload:
```
{
    "status": "success",
    "message": "Center created successfully",
    "data": {
        "name": "Nairobi",
        "id": "-LI6SDidpb7RS-1b4Tx8",
        "image": {
            "public_id": "cr4mxeqx5zb8rlakpfkg",
            "version": 1372275963,
            "signature": "63bfbca643baa9c86b7d2921d776628ac83a1b6e",
            "width": 864,
            "height": 576,
            "format": "jpg",
            "resource_type": "image",
            "created_at": "2017-06-26T19:46:03Z",
            "bytes": 120253,
            "type": "upload",
            "url": "https://res.cloudinary.com/demo/image/upload/v1372275963/cr4mxeqx5zb8rlakpfkg.jpg",
            "secure_url": "https://res.cloudinary.com/demo/image/upload/v1372275963/cr4mxeqx5zb8rlakpfkg.jpg"
        }
    }
}
```
#### Any background context you want to provide?
>- N/A
#### What are the relevant pivotal tracker stories?
>- [159192785](https://www.pivotaltracker.com/story/show/159192785)
#### Blocked Tickets
>- [159192999](https://www.pivotaltracker.com/story/show/159192999)
>- [159192910](https://www.pivotaltracker.com/story/show/159192910)
>- [159231737](https://www.pivotaltracker.com/story/show/159231737)
>- [159193135](https://www.pivotaltracker.com/story/show/159193135)
#### Screenshots (if appropriate)
>- N/A
#### Questions:
>- N/A
#### Postman Collection
>- [Link](https://www.getpostman.com/collections/de34759d20351de9a548)
